### PR TITLE
Localization

### LINKED ISSUE #13: Localization
### Idea

The website should be accessible to english and croatian speakers.

### Acceptance criteria

- [x] url scheme for hr/en
- [x] button for switching languages in appbar
- [x] make translation files load only once
- [x] translation files
- [x] translation strategy
- [x] ensure all views use translation for strings
- [x] remove `cspell` ignore

### Implementation

Resource files sound OK but Blazor has some strategies for URLs and possibly translation strings.

### PR DESCRIPTION
**Pull request by HrvojeJuric**
# Task

Fixes #13 

## Description

Done with embedded JSON file that switches depending on culture defined by URL scheme.

## Testing

Manual.

## Documentation

None. Will be added in #30.

### CODE REVIEW

File: src/Ozds.Client/Pages/DataMutatePage.razor
Comment (HrvojeJuric) [inline src/Ozds.Client/Pages/DataMutatePage.razor:1]: use enum here as well

File: src/Ozds.Client/App.razor
Comment (HrvojeJuric) [inline src/Ozds.Client/App.razor:1]: empty changes

File: src/Ozds.Client/Base/OzdsComponentLocalizer.cs
Comment (HrvojeJuric) [inline src/Ozds.Client/Base/OzdsComponentLocalizer.cs:1]: switch case is brittle - dont just check by string
using strings to check stuff like this is always very brittle - create static culture info or something like that that you will check against

Comment (HrvojeJuric) [reply src/Ozds.Client/Base/OzdsComponentLocalizer.cs:1]: also remove comment

File: src/Ozds.Client/Extensions/IServiceCollectionExtensions.cs
Comment (HrvojeJuric) [inline src/Ozds.Client/Extensions/IServiceCollectionExtensions.cs:1]: empty class pls remove

File: src/Ozds.Client/Shared/LocationsTable.razor [hunk 1]
```diff
@@ -139,6 +139,8 @@
 
   protected override async Task OnInitializedAsync()
   {
+    var a = T["Location"];
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/LocationsTable.razor:142]: what is this?

File: src/Ozds.Client/Shared/MeterGraph.razor [hunk 2]
```diff
@@ -175,17 +172,28 @@
...
+    };
+
     aggregateQueries = ScopedServices.GetRequiredService<OzdsAggregateQueries>();
     measureQueries = ScopedServices.GetRequiredService<OzdsMeasurementQueries>();
     _meter = await ScopedServices
       .GetRequiredService<OzdsMeterTableQueries>()
       .GetMeterById(Source);
 
+
+    if (Source.StartsWith("schneider"))
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/MeterGraph.razor:192]: we should do this via capabilities
there is a capabilities namespace in Ozds.Business
please add `HasApparentPower` property to `ICapabilities`, add implementation for abb and schneider capabilities and use _meter.Capcabilities.HasApparentPower here

File: src/Ozds.Client/Shared/MeterGraph.razor [hunk 3]
```diff
@@ -199,23 +207,23 @@
   {
     switch (type)
     {
-      case "Struja":
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/MeterGraph.razor:202]: again these things are brittle - please make an enum

File: src/Ozds.Client/Shared/NetUserRepDashboard.razor [hunk 4]
```diff
@@ -64,7 +64,7 @@
       <MudPaper Class="mud-width-full d-flex flex-column justify-space-between pa-4" Height="320px">
         <MudItem Class="" Style="font-size: large; font-weight: bold;">
           <MudText Typo="Typo.h5">
-            @(T["Potrošnja u tekućem mjesecu:"])
+            @(T["Current months consumption"] + ": ")
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/NetUserRepDashboard.razor:67]: use @(T["..."]):
dont add strings in c# when u can let razor handle it

File: src/Ozds.Client/Shared/NetUserRepDashboard.razor [hunk 5]
```diff
@@ -89,7 +89,7 @@
       <MudPaper Class="mud-width-full d-flex flex-column justify-space-between pa-4" Height="320px">
         <MudItem Class="" Style="font-size: large; font-weight: bold;">
           <MudText Typo="Typo.h5">
-            @(T["Potrošnja u prethodnom mjesecu:"])
+            @(T["Last months consumption"] + ": ")
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/NetUserRepDashboard.razor:92]: same as above

File: src/Ozds.Client/Shared/NetUserRepDashboard.razor [hunk 6]
```diff
@@ -100,12 +100,12 @@
         </MudContainer>
         <MudItem Class="" Style="font-size: large; font-weight: bold;">
           <MudText Typo="Typo.h6">
-            @(T["Iznos računa u prethodnom mjesecu:"])
+            @(T["Last months invoice total"] + ": ")
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/NetUserRepDashboard.razor:103]: samve as above

File: src/Ozds.Client/Shared/OperatorGraph.razor [hunk 7]
```diff
@@ -77,7 +77,7 @@
   <MudHidden Breakpoint="Breakpoint.SmAndDown">
     <MudPaper Class="d-flex align-center justify-center mud-width-full py-8 px-8 flex-column" Height="525px">
       <MudItem Class="d-flex justify-center align-center" Style="font-size: large; font-weight: bold;">
-        @(_dataType + " za: " + _title)
+        @(T[_dataType] + T[" for"] + ": " + _title)
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/OperatorGraph.razor:80]: again, let razor do this

File: src/Ozds.Client/Shared/OperatorGraph.razor [hunk 8]
```diff
@@ -265,13 +266,13 @@
   {
     if (source.StartsWith("abb"))
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/OperatorGraph.razor:267]: use capabilities as i described earlier

File: src/Ozds.Client/Shared/OperatorGraph.razor [hunk 9]
```diff
@@ -282,23 +283,23 @@
   {
     switch (type)
     {
-      case "Struja":
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/OperatorGraph.razor:285]: make enum please as described earlier

File: src/Ozds.Client/Shared/OperatorGraph.razor [hunk 10]
```diff
@@ -332,11 +333,11 @@
 
   private async Task GetValues(DateTimeOffset fromDate, DateTimeOffset toDate, bool reRenderChart = true)
   {
-    var quorterHourlyMeasures = await aggregateQueries.ReadAgnostic(new[] { "x => x.Interval == 0" }, fromDate, toDate, 1, 9000);
+    var quarterHourlyMeasures = await aggregateQueries.ReadAgnostic(new[] { "x => x.Interval == 0" }, fromDate, toDate, 1, 9000);
     var measures = await measureQueries.ReadAgnostic(new[] { $"x => x.MeterId == \"{_title}\"" }, fromDate, toDate, 1, 3000);
     switch (_dataType)
     {
-      case "Struja":
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/OperatorGraph.razor:339]: make enum please as described earlier

File: src/Ozds.Client/Shared/Print/NetworkUserCalculationPrintAbbreviations.razor [hunk 11]
```diff
@@ -1,26 +1,26 @@
...
 <p>
-  <strong>OIEiK</strong> – naknada za obnovljive izvore energije i
-  visokoučinkovitu kogeneraciju
+  <strong>@T["OIEiK"]</strong> – @T["fee for renewable energy sources and high-efficiency cogeneration"]
 </p>
-<p><strong>OMM</strong> – obračunsko mjerno mjesto</p>
-<p><strong>P</strong> - procjena</p>
-<p><strong>RJT</strong> – radna energija u doba u doba jedinstvene dnevne tarife</p>
+<p><strong>@T["OMM"]</strong> – @T["billing metering point"]</p>
+<p><strong>@T["P"]</strong> - @T["estimate"]</p>
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Shared/Print/NetworkUserCalculationPrintAbbreviations.razor:16]: this is my fault for leaving it here - we have no estimates right no so please change this line to
```html
<p><strong>@T["O"]</strong> - @T["read "]</p>
```

**Review by HrvojeJuric — COMMENTED** [review]
all in all good - just needs a few touch ups to reduce brittleness

**Comment by GabrijelAB** [comment]
Also fixed some strings in MeterGraph I forgot

File: src/Ozds.Client/App.razor
Comment (GabrijelAB) [reply src/Ozds.Client/App.razor:1]: neznam što ovo zanći

Comment (HrvojeJuric) [reply src/Ozds.Client/App.razor:1]: ovaj @code dolje nis ne radi i @injectovi isto nis ne rade

Comment (HrvojeJuric) [reply src/Ozds.Client/App.razor:1]: zas maknut HeadOutlet - to nam treba

File: src/Ozds.Client/App.razor [hunk 12]
```diff
@@ -1,10 +1,6 @@
-<HeadOutlet @rendermode="RenderMode.InteractiveServer"/>
```
Comment (HrvojeJuric) [inline src/Ozds.Client/App.razor:1]: ovo nam treba - nezz zas si maknuo headoutlet

File: src/Ozds.Client/Base/OzdsComponentLocalizer.cs [hunk 13]
```diff
@@ -1,9 +1,70 @@
...
-    get { return notLocalized; }
+    get
+    {
+      Dictionary<string, string> _activeTranslation = [];
+      var culture = CultureInfo.CurrentCulture.TwoLetterISOLanguageName;
+      var hrCulture = CultureInfo.CreateSpecificCulture("hr-HR")
+        .TwoLetterISOLanguageName;
+      var enCulture = CultureInfo.CreateSpecificCulture("en-US")
+        .TwoLetterISOLanguageName;
+      if (culture == enCulture)
```
Comment (HrvojeJuric) [inline src/Ozds.Client/Base/OzdsComponentLocalizer.cs:29]: zas ne sam CultureInfo.CurrentCulture == Culture.CreateSpecificCulture("hr-HR")?
i dolje isto tak za en?

**Review by HrvojeJuric — COMMENTED** [review]
jos par stvari

File: src/Ozds.Client/App.razor
Comment (GabrijelAB) [reply src/Ozds.Client/App.razor:1]: kaj? ovo su jedine promjene 
image

File: src/Ozds.Client/App.razor (hunk 12 above)
Comment (GabrijelAB) [reply src/Ozds.Client/App.razor:1]: aha to je već od prije. Dodao

**Review by HrvojeJuric — COMMENTED** [review]
lgtm

**Pull request merged**